# Patches for microb-engine
# Oleg Romashin <ext-oleg.2.romashin@nokia.com>
# DP: This patch should fix problem with stripped libc (uclibc) library
#     Parsing of proxy,http headers failed without this workaround

Index: mozilla/intl/uconv/native/nsNativeUConvService.cpp
===================================================================
--- mozilla.orig/intl/uconv/native/nsNativeUConvService.cpp
+++ mozilla/intl/uconv/native/nsNativeUConvService.cpp
@@ -142,21 +142,25 @@
 #ifdef DEBUG
     mFrom = from;
     mTo = to;
 #endif
 
     mConverter = iconv_open(to, from);
     if (mConverter == (iconv_t) -1 )    
     {
+        mConverter = iconv_open("ISO-8859-1", from);
+        if (mConverter == (iconv_t) -1 )
+        {
 #ifdef DEBUG
         printf(" * IConvAdaptor - FAILED Initing: %s ==> %s\n", from, to);
 #endif
         mConverter = nsnull;
         return NS_ERROR_FAILURE;
+        }
     }
     return NS_OK;
 }
 
 // From some charset to ucs2
 nsresult 
 IConvAdaptor::Convert(const char * aSrc, 
                      PRInt32 * aSrcLength, 
